tests/test_postgres_incremental_replication.py

```diff
@@ -249,369 +343,449 @@ def get_properties():
-             ('properties', 'our_money'): {'inclusion': 'available', 'selected-by-default': True, 'sql-datatype': 'money'}},
-            metadata.to_map(md))
-
+        # perform table selection
+        print('selecting {} and all fields within the table'.format(test_table_name))
+        schema_and_metadata = menagerie.get_annotated_schema(conn_id, test_catalog['stream_id'])
         additional_md = [{ "breadcrumb" : [], "metadata" : {'replication-method' : 'INCREMENTAL', 'replication-key' : 'OUR TS TZ'}}]
-        _ = connections.select_catalog_and_fields_via_metadata(conn_id, test_catalog,
-                                                               menagerie.get_annotated_schema(conn_id, test_catalog['stream_id']),
-                                                               additional_md)
+        _ = connections.select_catalog_and_fields_via_metadata(conn_id, test_catalog, schema_and_metadata, additional_md)
 
         # clear state
         menagerie.set_state(conn_id, {})
 
-        # Sync Job 1
+        # run sync job 1 and verify exit codes
         sync_job_name = runner.run_sync_mode(self, conn_id)
-
-        # verify tap and target exit codes
         exit_status = menagerie.get_exit_status(conn_id, sync_job_name)
         menagerie.verify_sync_exit_status(self, exit_status, sync_job_name)
 
-        record_count_by_stream = runner.examine_target_output_file(self,
-                                                                   conn_id,
-                                                                   self.expected_sync_streams(),
-                                                                   self.expected_pks())
-
-        self.assertEqual(record_count_by_stream, { test_table_name: 2})
+        # get records
+        record_count_by_stream = runner.examine_target_output_file(
+            self, conn_id, self.expected_sync_streams(), self.expected_primary_keys()
+        )
         records_by_stream = runner.get_records_from_target_output()
-
         table_version = records_by_stream[test_table_name]['table_version']
-        self.assertEqual(3, len(records_by_stream[test_table_name]['messages']))
-        self.assertEqual(records_by_stream[test_table_name]['messages'][0]['action'], 'activate_version')
-        self.assertEqual(records_by_stream[test_table_name]['messages'][1]['action'], 'upsert')
-        self.assertEqual(records_by_stream[test_table_name]['messages'][2]['action'], 'upsert')
-
-        # verifications about individual records
-        for table_name, recs in records_by_stream.items():
-            # verify the persisted schema was correct
-            self.assertEqual(recs['schema'],
-                             expected_schemas[table_name],
-                             msg="Persisted schema did not match expected schema for table `{}`.".format(table_name))
-
-        expected_record_2 = {'our_decimal': decimal.Decimal('1234567890.01'),
```

Comment:
I like that you moved the expected record right after the inserted record. It is easier to read and see what is happening